mirage/utils.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable max-statements-per-line */
+import {Response} from 'ember-cli-mirage';
+
+export function paginatedResponse(modelName) {
+    return function (schema, request) {
+        let page = +request.queryParams.page || 1;
+        let limit = request.queryParams.limit || 15;
+        let pages, next, prev, models;
+
+        let allModels = this.serialize(schema[modelName].all())[modelName];
+
+        if (limit === 'all') {
+            pages = 1;
+        } else {
+            limit = +limit;
+
+            let start = (page - 1) * limit;
+            let end = start + limit;
+
+            pages = Math.ceil(allModels.length / limit);
+            models = allModels.slice(start, end);
+
+            if (start > 0) {
+                prev = page - 1;
+            }
+
+            if (end < allModels.length) {
+                next = page + 1;
+            }
+        }
+
+        return {
+            meta: {
+                pagination: {
+                    page,
+                    limit,
+                    pages,
+                    total: allModels.length,
+                    next: next || null,
+                    prev: prev || null
+                }
+            },
+            [modelName]: models || allModels
+        };
+    };
+}
+
+export function maintenanceResponse() {
+    return new Response(503, {}, {
+        errors: [{
+            errorType: 'Maintenance'
+        }]
+    });
+}
+
+export function versionMismatchResponse() {
+    return new Response(400, {}, {
+        errors: [{
+            errorType: 'VersionMismatchError'
+        }]
+    });
+}
+
+// minimalist deparam implementation
+export function deparam(string) {
```

Comment:
@kevinansfield I added this and removed the `jquery-deparam` dependency because Mirage didn't have access to it for some reason (may have been because I was using it in dev and not test mode), but regardless this was enough for our tests, so I think it's probably good to stay.